Source/CSharpAnalytics/Serializers/AppDataContractSerializer.cs

```diff
@@ -48,36 +49,32 @@ private static string GetDefaultFolderPath()
         public static async Task<T> Restore<T>(string filename = null, bool deleteBadData = false)
         {
             var serializer = new DataContractSerializer(typeof(T), new[] { typeof(DateTimeOffset) });
+            var settings = new XmlReaderSettings { Async = true };
 
             try
             {
                 var file = GetFilePath<T>(filename);
-
                 try
                 {
-                    using (var inputStream = new FileStream(file, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, true))
+                    using (var fileStream = new FileStream(file, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, true))
+                    using (var xmlReader = XmlReader.Create(fileStream, settings))
                     {
-                        if (inputStream.Length == 0)
-                        {
-                            return default(T);
-                        }
-
-                        using (var memoryStream = new MemoryStream())
-                        {
-                            await inputStream.CopyToAsync(memoryStream);
-                            await inputStream.FlushAsync();
-                            memoryStream.Seek(0, SeekOrigin.Begin);
-
-                            return (T)serializer.ReadObject(memoryStream);
-                        }
```

Comment:
In matter of fact it doesn't. The previous format was a non-intended xml and by using XmlWriterSettings it produces an intended xml. The serializer (DataContractSerializer) didn't changed.